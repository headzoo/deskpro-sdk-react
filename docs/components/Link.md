Link
====
A link which dispatches to a route location when clicked.

## Props

```jsx
<Link
    to={string}
    params={object}
/>
```

## Examples

```jsx
<Link to="settings">
    Click
</Link>
<Link to="settings" params={{ country: 'uk' }}>
    Click
</Link>
```
