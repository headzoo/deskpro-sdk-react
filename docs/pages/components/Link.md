Overview
========
The `Link` component changes the current route location when clicked. See the [route documentation](/pages/props/#route) for more information.

## Props

```jsx
<Link
    to={string}
    params={object}
/>
```

## Example
The following example renders two links which change the current location when clicked.

```jsx
import React from 'react';
import { Link, sdkConnect } from 'deskpro-sdk-react';

class Page extends React.Component {
    render() {
        return (
            <div>
                <Link to="settings">
                    Click
                </Link>
                <Link to="settings" params={{ country: 'uk' }}>
                    Click
                </Link>
            </div>
        );
    }
}

export default sdkConnect(Page);
```

!!! note
    Components using the `Link` component must be [connected to the SDK](/pages/props/#connecting-your-components) using `sdkConnect` or `DeskproSDK`.
