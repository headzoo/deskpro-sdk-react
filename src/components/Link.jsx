import React from 'react';
import PropTypes from 'prop-types';
import { propKeyFilter } from '../utils/props';

/**
 * A link which dispatches to a route location when clicked.
 *
 * Example:
 *
 * ```jsx
 * render() {
 *  return (
 *    <div>
 *      <Link to="index">Index Page</Link>
 *    </div>
 *  );
 * }
 * ```
 */
class Link extends React.Component {
  static propTypes = {
    /**
     * Route that will be dispatched when the button is clicked.
     */
    to: PropTypes.string,

    /**
     * Params passed with the route.
     */
    params: PropTypes.object,

    /**
     * Children rendered into the button.
     */
    children: PropTypes.node,

    /**
     * Called when the button is clicked.
     */
    onClick: PropTypes.func
  };

  /**
   * Specifies the default values for props
   */
  static defaultProps = {
    to:       '',
    params:   {},
    children: '',
    onClick:  () => {}
  };

  /**
   * Context values the child wants passed down from the parent
   */
  static contextTypes = {
    route: PropTypes.object
  };

  /**
   * Called when the link is clicked
   *
   * Dispatches to the 'to' route unless the default is prevented.
   *
   * @param {Event} e
   */
  handleClick = (e) => {
    this.props.onClick(e);
    if (!e.defaultPrevented) {
      this.context.route.to(this.props.to, this.props.params);
      e.preventDefault();
    }
  };

  /**
   * @returns {XML}
   */
  render() {
    const { children, ...props } = this.props;

    return (
      <a {...propKeyFilter(props, Link.propTypes)} onClick={this.handleClick}>
        {children}
      </a>
    );
  }
}

export default Link;
