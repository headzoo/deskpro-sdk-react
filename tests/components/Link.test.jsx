import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../../src/components/Link';

it('Link renders anchor tag', () => {
  const renderedValue = renderer.create(
    <Link to="index">Index</Link>
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
