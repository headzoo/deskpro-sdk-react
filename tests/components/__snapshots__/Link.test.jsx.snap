// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders anchor tag 1`] = `
<a
  onClick={[Function]}
>
  Index
</a>
`;
